src/import/sst_service.rs

```diff
@@ -368,6 +368,45 @@ where
         self.threads.spawn_ok(handle_task);
     }
 
+    // Downloads KV file and performs key-rewrite then apply kv into this tikv store.
+    fn apply(&mut self, _ctx: RpcContext<'_>, req: ApplyRequest, sink: UnarySink<ApplyResponse>) {
+        let label = "apply";
+        let timer = Instant::now_coarse();
+        let importer = Arc::clone(&self.importer);
+        let engine = self.engine.clone();
+        let limiter = self.limiter.clone();
+        let start = Instant::now();
+
+        let handle_task = async move {
+            // Records how long the apply task waits to be scheduled.
+            sst_importer::metrics::IMPORTER_APPLY_DURATION
+                .with_label_values(&["queue"])
+                .observe(start.saturating_elapsed().as_secs_f64());
```

Comment:
Maybe record time cost of other steps too?

Reply:
Done, PTAL again